providers/google/src/airflow/providers/google/cloud/transfers/sftp_to_gcs.py

```diff
@@ -166,16 +173,22 @@ def _copy_single_object(
             destination_object,
         )
 
-        with NamedTemporaryFile("w") as tmp:
-            sftp_hook.retrieve_file(source_path, tmp.name, prefetch=self.sftp_prefetch)
-
-            gcs_hook.upload(
-                bucket_name=self.destination_bucket,
-                object_name=destination_object,
-                filename=tmp.name,
-                mime_type=self.mime_type,
-                gzip=self.gzip,
-            )
+        if self.use_stream:
+            dest_bucket = gcs_hook.get_bucket(self.destination_bucket)
+            dest_blob = dest_bucket.blob(destination_object)
```

Comment:
Since we are calling the `get_bucket` method and then calling the `.blob` method on the bucket object, do you agree that we can create a method called `blob` on the hook side to return the blob directly like the following:

```python
   # hooks/gcs.py
    def blob(self, bucket_name: str, object_name: str, user_project: str | None = None):
            bucket = self.get_bucket(self.destination_bucket, user_project)
            return bucket.blob(object_name)
```
And then we can call it like the following here:
```suggestion
            dest_blob = gcs_hook.blob(self.destination_bucket, destination_object)
```

Also, if we implement the `blob` method in the GCSHook class then we do not need to add the public `get_bucket` method.

Reply:
Replied above